ts/packages/aiclient/src/openai.ts

```diff
@@ -518,11 +528,20 @@ function createAzureOpenAIChatModel(
                         if (data.choices && data.choices.length > 0) {
                             const delta = data.choices[0].delta?.content ?? "";
                             if (delta) {
+                                fullResponseText += delta;
                                 yield delta;
                             }
                         }
                         if (data.usage) {
```

Comment:
If the API doesn't report usage (or usage reporting has been disabled), it's possible that the request won't get logged.  I would move this to the IF above:

                    if (evt.data === "[DONE]") {
                        break;
                    }

Reply:
Great catch, thank you! I moved the code and confirmed that it works, with and without token usage enabled.